src/MasonryLayout/MasonryLayout.tsx

```diff
@@ -236,7 +240,10 @@ export default ({
                     hiddenChildren={hiddenChildren}
                     width={masonryLayoutRef.current?.clientWidth ?? -1}
                     columns={columns}
-                    minWidth={minWidth}
+                    minWidth={Math.min(
+                        minWidth,
+                        masonryLayoutRef.current?.clientWidth ?? minWidth
+                    )}
```

Comment:
This could potentially also be extracted to a function, in order to better communicate the intention of using Math.min(...). It's kind of difficult to keep track of what is the purpose of all the different variables.

E.g. `getMinimumWidthToContainContent()`

I may be wrong though 🤔